tests/components/zone/test_init.py

```diff
@@ -243,3 +238,252 @@ def test_in_zone_works_for_passive_zones(self):
+    assert state_2.attributes["latitude"] == 3
+    assert state_2.attributes["longitude"] == 4
+    assert state_3 is None
+    assert len(ent_reg.entities) == 0
+
+    with patch(
+        "homeassistant.config.load_yaml_config_file",
+        autospec=True,
+        return_value={
+            DOMAIN: [
+                {"name": "yaml 2", "latitude": 3, "longitude": 4},
+                {"name": "yaml 3", "latitude": 5, "longitude": 6},
+            ]
+        },
+    ):
+        with pytest.raises(Unauthorized):
+            await hass.services.async_call(
+                DOMAIN,
+                SERVICE_RELOAD,
+                blocking=True,
+                context=Context(user_id=hass_read_only_user.id),
+            )
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_RELOAD,
+            blocking=True,
+            context=Context(user_id=hass_admin_user.id),
+        )
+        await hass.async_block_till_done()
+
+    assert count_start + 3 == len(hass.states.async_entity_ids())
+
+    state_1 = hass.states.get("zone.yaml_1")
+    state_2 = hass.states.get("zone.yaml_2")
+    state_3 = hass.states.get("zone.yaml_3")
+
+    assert state_1 is None
+    assert state_2 is not None
+    assert state_2.attributes["latitude"] == 3
+    assert state_2.attributes["longitude"] == 4
+    assert state_3 is not None
+    assert state_3.attributes["latitude"] == 5
+    assert state_3.attributes["longitude"] == 6
+
+
+async def test_load_from_storage(hass, storage_setup):
+    """Test set up from storage."""
+    assert await storage_setup()
+    state = hass.states.get(f"{DOMAIN}.from_storage")
+    assert state.state == "zoning"
+    assert state.name == "from storage"
+    assert state.attributes.get(ATTR_EDITABLE)
+
+
+async def test_editable_state_attribute(hass, storage_setup):
+    """Test editable attribute."""
+    assert await storage_setup(
+        config={DOMAIN: [{"name": "yaml option", "latitude": 3, "longitude": 4}]}
+    )
+
+    state = hass.states.get(f"{DOMAIN}.from_storage")
+    assert state.state == "zoning"
+    assert state.attributes.get(ATTR_FRIENDLY_NAME) == "from storage"
+    assert state.attributes.get(ATTR_EDITABLE)
+
+    state = hass.states.get(f"{DOMAIN}.yaml_option")
+    assert state.state == "zoning"
+    assert not state.attributes.get(ATTR_EDITABLE)
+
+
+async def test_ws_list(hass, hass_ws_client, storage_setup):
+    """Test listing via WS."""
+    assert await storage_setup(
+        config={DOMAIN: [{"name": "yaml option", "latitude": 3, "longitude": 4}]}
+    )
+
+    client = await hass_ws_client(hass)
+
+    await client.send_json({"id": 6, "type": f"{DOMAIN}/list"})
+    resp = await client.receive_json()
+    assert resp["success"]
+
+    storage_ent = "from_storage"
+    yaml_ent = "from_yaml"
+    result = {item["id"]: item for item in resp["result"]}
+
+    assert len(result) == 1
+    assert storage_ent in result
+    assert yaml_ent not in result
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4870504)